libs/auth-api-lib/src/lib/clients/admin/admin-clients.service.ts

```diff
@@ -67,24 +92,78 @@ export class AdminClientsService {
     const client = await this.clientModel.create({
       clientId: clientDto.clientId,
       clientType: clientDto.clientType,
+      domainName: tenantId,
       nationalId: tenant.nationalId,
       clientName: clientDto.clientName,
       ...this.defaultClientAttributes(clientDto.clientType),
     })
-    return this.formatClient(client)
+    return this.findByTenantIdAndClientId(tenantId, client.clientId)
   }
 
-  defaultClientAttributes(clientType: AdminClientType) {
+  defaultClientAttributes(clientType: ClientType) {
     switch (clientType) {
       default:
         return {}
     }
   }
 
-  formatClient(client: Client): AdminClientDto {
+  private formatClient(
+    client: Client,
+    translations?: Map<string, string>,
+  ): AdminClientDto {
+    const enDisplayName = translations?.get('clientName')
+
     return {
       clientId: client.clientId,
       clientType: client.clientType,
+      tenantId: client.domainName ?? '',
+      displayName: client.clientName
+        ? [
+            { locale: 'is', value: client.clientName },
+            ...(enDisplayName
+              ? [
+                  {
+                    locale: 'en',
+                    value: enDisplayName,
+                  },
+                ]
+              : []),
+          ]
+        : [],
+      absoluteRefreshTokenLifetime: client.absoluteRefreshTokenLifetime,
+      slidingRefreshTokenLifetime: client.slidingRefreshTokenLifetime,
+      refreshTokenExpiration: client.refreshTokenExpiration,
+      supportsCustomDelegation: client.supportsCustomDelegation,
+      supportsLegalGuardians: client.supportsLegalGuardians,
+      supportsProcuringHolders: client.supportsProcuringHolders,
+      supportsPersonalRepresentatives: client.supportsPersonalRepresentatives,
+      promptDelegations: client.promptDelegations,
+      requireApiScopes: client.requireApiScopes,
+      requireConsent: client.requireConsent,
+      allowOfflineAccess: client.allowOfflineAccess,
+      requirePkce: client.requirePkce,
+      accessTokenLifetime: client.accessTokenLifetime,
+      redirectUris: client.redirectUris?.map((uri) => uri.redirectUri) ?? [],
+      postLogoutRedirectUris:
+        client.postLogoutRedirectUris?.map((uri) => uri.redirectUri) ?? [],
+      supportTokenExchange:
+        client.allowedGrantTypes?.some(
+          (grantType) => grantType.grantType === GrantTypeEnum.TokenExchange,
+        ) ?? false,
+      customClaims:
+        client.claims?.reduce(
+          (acc, curr) => ({ ...acc, [curr.type]: curr.value }),
+          {},
+        ) ?? {},
```

Comment:
Nonono. 🤓 Try [`Object.fromEntries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries)

Reply:
Though it's probably better (eg for the GraphQL schema) to expose this as an array of claim objects instead of as an arbitrary JSON object.